src/west/manifest.py

```diff
@@ -1648,14 +1648,28 @@ def _load_project(self, pd: Dict, url_bases: Dict[str, str],
                       west_commands=pd.get('west-commands'),
                       topdir=self.topdir, remote_name=remote)
 
-        if self.topdir:
-            assert isinstance(ret.abspath, str)
-            apath = Path(ret.abspath)
-            topdir = Path(self.topdir)
-            if escapes_directory(apath, topdir) or apath == topdir:
-                self._malformed(f'project {name} absolute path {apath} '
-                                'is not a subdirectory of topdir ' +
-                                self.topdir)
+        # Make sure the return Project's path does not escape the
+        # workspace. We can't use escapes_directory() as that
+        # resolves paths, which has proven to break some existing
+        # users who use symlinks to existing project repositories
+        # outside the workspace as a cache.
+        #
+        # Instead, normalize the path and make sure it's neither
+        # absolute nor starts with a '..'. This is intended to be
+        # a purely lexical operation which should therefore ignore
+        # symbolic links.
+        ret_norm = os.path.normpath(ret.path)
+
+        if os.path.isabs(ret_norm):
+            self._malformed(f'project "{ret.name}" has absolute path '
+                            f'{ret.path}; this must be relative to the '
+                            f'workspace topdir' +
+                            (f' ({self.topdir})' if self.topdir else ''))
+
+        if ret_norm.startswith('..'):
```

Comment:
so what if user does: `path-prefix: dummy/../../../i_escaped` ?

Reply:
This case is shown above in the example errors.

Reply:
(And now has an explicit test case apart from the existing path-prefix escape check)